File: IOS/appPickerViewConversor/appPickerViewConversor/ViewController.m
//
//  ViewController.m
//  appPickerViewConversor
//
//  Created by Jorge on 11/04/24.
//

#import "ViewController.h"

@interface ViewController ()
{
    NSArray *paises;
    NSArray *banderas;
}
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    paises = @[ @"Australia", @"China", @"Francia", @"Inglaterra", @"Japón", @"México", @"UE" ];
    
    banderas = @[ @"🇦🇺", @"🇨🇳", @"🇫🇷", @"🏴󠁧󠁢󠁥󠁮󠁧󠁿", @"🇯🇵", @"🇲🇽", @"🇪🇺" ];
    
    _tipoDeCambio = [ [ NSArray alloc ] initWithObjects:[ NSNumber numberWithFloat:1.54], [ NSNumber numberWithFloat:7.27 ], [ NSNumber numberWithFloat:6.5595 ], [ NSNumber numberWithFloat:0.80 ], [ NSNumber numberWithFloat:153.13 ], [ NSNumber numberWithFloat:16.67 ], [ NSNumber numberWithFloat:0.94 ], nil ];
    
    self.pickerPaises.delegate = self;
    self.pickerPaises.dataSource = self;
    
    _etiquetaResultado.text = @"Equivalencia de divisas";
}

-(NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView {
    
    return 1;
    
}

- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
    
    return [paises count];
    
}

- (NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component {
    return paises[row];
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component {
    float valorMoneda = [ [ _tipoDeCambio objectAtIndex:row ] floatValue ];
    
    float dolares = [ _cantidadDolares.text floatValue ];
    
    float resultado = valorMoneda * dolares;
    NSString *resultadoString = [ [ NSString alloc ] initWithFormat:@"%.2f USD = %.2f %@",dolares,resultado, [banderas objectAtIndex:row ] ];
    
    NSLog(@"Resultado: %@", resultadoString);
    
    _etiquetaResultado.text = resultadoString;
    
}


@end

File: IOS/SWIFT_IOS/appCrearPickerView/appCrearPickerView/ViewController.swift
//
//  ViewController.swift
//  appCrearPickerView
//
//  Created by Jorge Cisneros on 23/05/24.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var texto1: UITextField!
    
    let banderas = ["Mexico 🇲🇽", "USA 🇺🇸", "Francia 🇫🇷", "Ukrania 🇺🇦", "Cuba 🇨🇺", "Bahamas 🇧🇸"]
    
    var banderaSeleccionada : String?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        crearPickerViewBanderas()
        crearToolBar()
        
    }

    func crearPickerViewBanderas() {
        let banderasPicker = UIPickerView()
        
        banderasPicker.delegate = self
        texto1.inputView = banderasPicker
    }
    
    func crearToolBar() {
        let toolBar = UIToolbar()
        toolBar.sizeToFit()
        toolBar.barTintColor = .green
        toolBar.tintColor = .black
        
        let botonOcultar = UIBarButtonItem(title: "⬇", style: .plain, target: self, action: #selector(ViewController.dismissKeyboard))
        
        toolBar.setItems([botonOcultar], animated: true)
        
        let botonAdios = UIBarButtonItem(title: "👋", style: .plain, target: self, action: #selector(ViewController.adios))
        
        toolBar.setItems([botonAdios], animated: false)
        
        let espacio = UIBarButtonItem(barButtonSystemItem: UIBarButtonItem.SystemItem.flexibleSpace, target: nil, action: nil)
        
        self.toolbarItems = [botonOcultar, espacio, botonAdios]
        toolBar.setItems(toolbarItems, animated: false)
        texto1.inputAccessoryView = toolBar
    }
    
    @objc func dismissKeyboard() {
        view.endEditing(true)
    }
    
    @objc func adios() {
        
        //UIControl().sendAction(#selector(NSXPCConnection.suspend), to: UIApplication.shared, for: nil)
        
        UIControl().sendAction(#selector(NSXPCConnection.suspend), to: UIApplication.shared, for: nil)
        
    }

}// FIN DEL VIEW CONTROLER

extension CGFloat {
    
    static func random() -> CGFloat {
        
        return CGFloat(arc4random()) / CGFloat(UInt32.max)
        
    }
    
}

extension UIColor {
    
    static func random() -> UIColor {
        
        return UIColor(red: .random(), green: .random(), blue: .random(), alpha: 1.0)
        
    }
    
}

extension ViewController : UIPickerViewDelegate, UIPickerViewDataSource {
    
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
        
        return 1
    
    }
    
    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        
        return banderas.count
    
    }
    
    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
        
        return banderas[ row ]
    
    }
    
    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
        
        banderaSeleccionada = banderas[ row ]
        texto1.backgroundColor = UIColor.random()
        texto1.text = banderaSeleccionada
        self.view.backgroundColor = UIColor.random()
        
    }
    
    func pickerView(_ pickerView: UIPickerView, viewForRow row: Int, forComponent component: Int, reusing view: UIView?) -> UIView {
        
        var label : UILabel
        
        if let view = view as? UILabel {
            
            label = view
            
        } else {
            
            label = UILabel()
            
        }
        
        label.textColor = UIColor.random()
        label.textAlignment = .center
        label.font = UIFont( name: "Verdana", size: 35 )
        label.text = banderas[ row ]
        
        return label
        
    }
    
}

